### PR TITLE
test: full coverage on core libs

### PR DESCRIPTION
**Pull request by goga-m**
## Summary
Full coverage for lib core modules for https://app.clickup.com/t/86e0bj67k

## Checklist

- [ ] My changes look good in both light AND dark mode
- [ ] The change is not hardcoded to a single network, but has multi-asset in mind
- [ ] I checked my changes for obvious issues, debug statements and commented code
- [ ] Documentation _(if necessary)_
- [ ] Tests _(if necessary)_
- [ ] Ready to be merged

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -989,10 +989,10 @@ jobs:
...
         env:
             COVERAGE_INCLUDE_PATH: src/app/lib/profiles,src/app/lib/mainsail,src/app/lib/intl,src/app/lib/markets
-            COVERAGE_THRESHOLD_LINES: 93.76
-            COVERAGE_THRESHOLD_BRANCHES: 87.93
-            COVERAGE_THRESHOLD_FUNCTIONS: 93.47
-            COVERAGE_THRESHOLD_STATEMENTS: 93.55
+            COVERAGE_THRESHOLD_LINES: 99.85
+            COVERAGE_THRESHOLD_BRANCHES: 99.78
+            COVERAGE_THRESHOLD_FUNCTIONS: 99.69
+            COVERAGE_THRESHOLD_STATEMENTS: 98.6
```
Comment (goga-m) [inline .github/workflows/test.yml:995]: @shahin-hq @ItsANameToo the few lines that were left uncovered require refactor, they are some edge cases that we might not need, and as the PR grew in size, I focused here only to cover without code refactors. We could investigate to simplify.

**Review by ItsANameToo — APPROVED** [review]
